File: src/main/java/com/procurement/evaluation/model/entity/AwardEntity.java
package com.procurement.evaluation.model.entity;

import java.util.UUID;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.Column;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;

@Getter
@Setter
@Table("evaluation_award")
public class AwardEntity {

    @PrimaryKeyColumn(name = "cp_id", type = PrimaryKeyType.PARTITIONED)
    private String cpId;

    @PrimaryKeyColumn(name = "stage", type = PrimaryKeyType.CLUSTERED)
    private String stage;

    @PrimaryKeyColumn(value = "token_entity", type = PrimaryKeyType.CLUSTERED)
    private UUID token;

    @Column(value = "status")
    private String status;

    @Column(value = "status_details")
    private String statusDetails;

    @Column(value = "owner")
    private String owner;

    @Column(value = "json_data")
    private String jsonData;
}

File: src/main/kotlin/com/procurement/evaluation/model/entity/AwardEntity.kt
package com.procurement.evaluation.model.entity

import com.procurement.evaluation.domain.model.Owner
import com.procurement.evaluation.exception.ErrorException
import com.procurement.evaluation.exception.ErrorType
import java.util.*

data class AwardEntity(

    val cpId: String,

    val stage: String,

    val token: UUID,

    val status: String,

    var statusDetails: String,

    val owner: String,

    var jsonData: String
) {
    fun checkOwner(owner: Owner) {
        if (this.owner != owner.toString()) throw ErrorException(error = ErrorType.OWNER)
    }
}
